Update Command line tutorial to match 4.0.alpha11 `--help` output

nabware opened the pull request:
Updated docs to reflect implementation of `--headless` argument which replaces `--no-window`. 

godotengine/godot#62828

Calinou commented:
The full list of CLI arguments needs to be updated in general. Check the output of `--help` in Godot 4.0.alpha11. It's probably worth consoliating this into a single pull request.

nabware commented:
Updated Command line reference section to reflect Godot 4.0.alpha11 `--help` output.

- Added code formatting to text in descriptions that use `--argument` or `<argument>`
- Removed redundant placeholder arguments (e.g. `-l <locale>, --language <locale>` to `-l, --language <locale>`)
- Updated Thread model reference path from 'rendering/threads/thread_model' to 'rendering/**driver**/threads/thread_model'

Description for `--language` states `<locale>` is a two-letter code. However [Locale codes](https://docs.godotengine.org/en/latest/tutorials/i18n/locales.html#doc-locales) has three-letter codes. Didn't change as the `--help` output states the same thing.

nabware commented on tutorials/editor/command_line_tutorial.rst:
```diff
@@ -50,17 +50,20 @@ Command line reference
+| ``--quit``                               | Quit after the first iteration.                                                                                                                              |
 +------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
-| ``--path <directory>``                   | Path to a project (<directory> must contain a 'project.godot' file).                                                                                         |
+| ``-l``, ``--language <locale>``          | Use a specific locale (``<locale>`` being a two-letter code). See :ref:`doc_locales` for more details.                                                       |
```
I can change it to:

"Use a specific locale. `<locale>` follows the format `language_Script_COUNTRY_VARIANT` where language is a 2 or 3-letter language code in lower case and the rest is optional. See :ref:`doc_locales` for more details."

The `--help` output still says "is a two-letter code." in alpha11. Not sure how to get that updated.

bruvzg replied:
> The --help output still says "is a two-letter code." in alpha11. Not sure how to get that updated.

It's defined in the `main.cpp`: https://github.com/godotengine/godot/blob/9b3972ad48393559f772151646304e28f938a6fa/main/main.cpp#L295

Calinou reviewed (approved):
Tested locally, it works. Thanks!
